src/components/power-sensor/client.ts

```diff
@@ -71,31 +71,38 @@ export class PowerSensorClient implements PowerSensor {
       powersensorService.getPower.bind(powersensorService),
       request
     );
+
     return response.getWatts();
   }
 
   async getReadings(extra = {}) {
-    const readingFunctions = {
-      voltage: this.getVoltage.bind(this),
-      current: this.getCurrent.bind(this),
-      power: this.getPower.bind(this),
-    };
-
-    const tasks = Object.entries(readingFunctions).flatMap(([field, func]) =>
-      (async () => {
-        try {
-          return [[field, await func(extra)]];
-        } catch (error) {
-          if (!(error as Error).message.includes('Unimplemented')) {
-            throw error;
-          }
-          return [];
-        }
-      })()
-    );
-    const entries = await Promise.all(tasks);
-
-    return Object.fromEntries(entries.flat()) as PowerSensorReadings;
+    const readings: Record<string, any> = {};
+    try {
+      const ret = await this.getVoltage(extra);
+      readings['voltage'] = ret[0];
+      readings['isAc'] = ret[1];
```

Comment:
[nit] (same for current and power below)

```suggestion
      const [voltage, isAc] = await this.getVoltage(extra);
      readings.voltage = voltage;
      readings.isAc = isAc;
```